java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java

```diff
@@ -1337,8 +1337,10 @@ public static void addErrataToChannel(Set<Long> eids, Long cid) {
     public static ContentSource findVendorContentSourceByRepo(String repoUrl) {
         Criteria criteria = getSession().createCriteria(ContentSource.class);
         criteria.add(Restrictions.isNull("org"));
-        if (repoUrl.contains("mirrorlist.centos.org") || repoUrl.contains("mirrors.aliyun.com") ||
-                repoUrl.contains("mirrors.cloud.aliyuncs.com")) {
+        if (repoUrl.contains("mirrorlist.centos.org") ||
+                repoUrl.contains("mirrors.aliyun.com") ||
+                repoUrl.contains("mirrors.cloud.aliyuncs.com") ||
+                repoUrl.contains("amazonaws.com")) {
```

Comment:
This is not really wrong, but maybe not needed. Why we do this for centos is for the following reason:

centos urls have just the hostname and use query parameter to find the right distro in the mirror list.
In the `else` we drop the query parameter as we do not want to compare with the authtoken.

AFAIK amazon linux also uses mirror list, but it code the distro, version and arch components in the URL path.
This means the `else` tree would work as well.

I am not sure about the the other two.
But as I said, this does not make the code here wrong, just more complex than needed